pkg/engine/availabilitysets.go

```diff
@@ -11,6 +11,13 @@ import (
 	"github.com/Azure/go-autorest/autorest/to"
 )
 
+// PlatformFaultDomainCountUnset indicates an ARM expression should replace this value.
+const PlatformFaultDomainCountNotSet = 0
+
+// PlatformFaultDomainCount is the global value used to populate the fault domain count
+// for a VMAS. It may be changed when upgrade inspects an existing cluster.
+var PlatformFaultDomainCount int32 = PlatformFaultDomainCountNotSet
```

Comment:
This variable is being (ab)used as an API to tell template generation what value to set for all availability sets' `platformFaultDomainCount`. It is set by upgrade and scale: otherwise the default ARM expression might calculate a value that differs with an existing availability set's value, which is a deployment error.

It would be preferable to carry this context in the `api.ContainerService` itself if that is possible without polluting it.